test.sh

```diff
@@ -1,6 +1,16 @@
 #!/bin/bash -e
 
 readonly compose_file='docker-compose.test.yml'
+readonly v5_puppet_gem="~> 5.5.21"
+readonly v5_output_xml="spec/output/rspec_puppet5.xml"
+
+if [ "$#" -gt 0 ] && [ "$1" =  "5" ]; then
+  echo "Using Puppet v5 ('$v5_puppet_gem') gem for testing"
+  export PUPPET_VERSION=$v5_puppet_gem
+  export CI_SPEC_OPTIONS="-f RspecJunitFormatter -o '$v5_output_xml' -f progress"
```

Comment:
Just a quick check, and for my own edification... these `CI_SPEC_OPTIONS` are only required for Puppet V5?

Reply:
Correct. The default v6 empty `CI_SPEC_OPTIONS ` string will do pretty much the exact thing but output the results to `spec/output/rspec.xml` instead. Sadly, you can't just specify the output file but have to override the whole formatters config.